common/authentication/postgresql/metadata.go

```diff
@@ -120,7 +129,8 @@ func (m *PostgresAuthMetadata) GetPgxPoolConfig() (*pgxpool.Config, error) {
 		// This is because tokens expire, and connections can drop and need to be re-established at any time
 		// Fortunately, we can do this with the "BeforeConnect" hook
 		config.BeforeConnect = func(ctx context.Context, cc *pgx.ConnConfig) error {
-			at, err := tokenCred.GetToken(ctx, policy.TokenRequestOptions{
+			var at exported.AccessToken
+			at, err = tokenCred.GetToken(ctx, policy.TokenRequestOptions{
```

Comment:
In this specific case, the old code was correct.

This is inside a "lambda", so we should use an `err` that is scoped within the function.

The current code, however, uses `err` from the scope of the parent. That can cause unexpected issues if the method is invoked concurrently

```suggestion
			at, err := tokenCred.GetToken(ctx, policy.TokenRequestOptions{
```